File: structure/array/array.jl

# create array
a = [1,2,3,4]

# 增删改查
# add item
push!(a, 5) # inplace
insert!(a, 1, 6)
println(a)

# remove item
pop!(a)
deleteat!(a, 1)
println(a)

# assign, index start from 1
a[1] = 4
println(a)
println(a[1])

# iter
# 1. for each
for i in a
    println(i)
end

# 2. for idx
l = length(a)
for i in 1:l
    println(a[i])
end

File: structure/array/array.rs



fn main() {
    // 不可变长度
    // 初始化
    let a: [i32; 5] = [1; 5];
    let a: [i32; 5] = [1, 2, 3, 4, 5];
    // 长度
    println!("{}", a.len());
    // 索引
    println!("{}", a[0]);
    println!("{:?}", a.get(100));
    // 切片
    println!("{:?}", &a[0..2]);
    // 转vec
    println!("{:?}", a.to_vec());

    // 可变长度
    // 初始化
    let mut v: Vec<i32> = Vec::new();
    let mut v: Vec<i32> = vec![1, 2, 3, 4, 5];
    // 长度
    println!("{}", v.len());
    // 索引
    println!("{}", v[0]);
    println!("{:?}", v.get(100));
    // 增删改查
    v.push(6);
    println!("{:?}", v.pop());
    v[3] = 100;
    // 遍历
    for i in &v { // 控制权限
        println!("{}", i);
    }

    for i in &mut v {
        *i += 1;
    }

    for i in v.iter() {
        println!("{}", i);
    }

    v.iter().map(|x| println!("{}", x));
}